backend/assignments/1/src/main/java/org/example/TransactionReader.java

```
@@ -0,0 +1,75 @@
+package org.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class TransactionReader {
+    public static List<Transaction> readTransactionsFromFile(String filePath) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode root = objectMapper.readTree(reader);
+
+            return StreamSupport.stream(root.spliterator(), false)
+                    .map(TransactionReader::createTransactionFromNode)
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+    
+
+    private static Transaction createTransactionFromNode(JsonNode transactionNode) {
+        String type = transactionNode.get("type").asText();
+
+        switch (type) {
+            case "BUY":
+                return createBuyTransaction(transactionNode.get("data"));
+            case "UPDATE_PRICE":
+                return createUpdatePriceTransaction(transactionNode.get("data"));
+            case "ADD_VOLUME":
+                return createAddVolumeTransaction(transactionNode.get("data"));
+            case "SELL":
+                return createSellTransaction(transactionNode.get("data"));
+            default:
+                return null;
```

Lets not return `null` here. Lets throw an exception and catch it.